server.js

```diff
@@ -0,0 +1,99 @@
+const http = require('http');
+const fileSystem = require('fs');
+
+let _404;
+fileSystem.readFile('public/404.html', 'utf-8', (err, data) => {
```

Comment:
This is an asynchronous call so you should actually be doing this in the request event below.